service/telemetry.go

```diff
@@ -59,7 +59,7 @@ const (
 
 func init() {
 	//nolint:staticcheck
-	featuregate.Register(featuregate.Gate{
+	featuregate.GetRegistry().Register(featuregate.Gate{
```

Comment:
To see the benefit of this change, we should split init into something like:

```golang
func init() {
	initWithRegistry(featuregate.GetRegistry())
}

func initWithRegistry(registry *featuregate.Registry) {
	registry.Register(featuregate.Gate{
		ID:          useOtelForInternalMetricsfeatureGateID,
		Description: "controls whether the collector to uses open telemetry for internal metrics",
		Enabled:     configtelemetry.UseOpenTelemetryForInternalMetrics,
	})
}
```

Then change all the tests to run `initWithRegistry` and test using a custom "Registry" per test. Does it make sense?

Reply:
HI @bogdandrutu 

We add a new func 'newRegistry()' to create object in test.
and add the func 'initWithRegistry' with unit test.

Would you please review it?